src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -146,8 +146,12 @@ public void updateUserPassword(String username, String password) {
     }
 
     @Override
-    public void updateUserEmail(String username, String email) {
-        userRepository.saveUserEmailByUsername(username, email);
+    public void updateUserEmail(String username, String email, UUID codeId) {
+        var emailConfirmationCode = new EmailConfirmationCode(codeId, numberOfHoursToKeepEmailConformationCode);
```

Comment:
You don't need to create that class here. Just set the values for the DB request directly